pkg/volume/gce_pd/gce_util.go

```diff
@@ -80,17 +80,38 @@ func (gceutil *GCEDiskUtil) CreateVolume(c *gcePersistentDiskProvisioner) (strin
 	// GCE works with gigabytes, convert to GiB with rounding up
 	requestGB := volume.RoundUpSize(requestBytes, 1024*1024*1024)
 
-	// The disk will be created in the zone in which this code is currently running
-	// TODO: We should support auto-provisioning volumes in multiple/specified zones
-	zones, err := cloud.GetAllZones()
-	if err != nil {
-		glog.V(2).Infof("error getting zone information from GCE: %v", err)
-		return "", 0, nil, err
+	// Apply ProvisionerParameters (case-insensitive). We leave validation of
+	// the values to the cloud provider.
+	diskType := ""
+	zone := ""
+	for k, v := range c.options.ProvisionerParameters {
+		switch strings.ToLower(k) {
+		case "type":
+			diskType = v
+		case "zone":
+			zone = v
+		default:
+			return "", 0, nil, fmt.Errorf("invalid option %q for volume plugin %s", k, c.plugin.GetPluginName())
+		}
 	}
 
-	zone := volume.ChooseZoneForVolume(zones, c.options.PVCName)
+	// TODO: implement c.options.ProvisionerSelector parsing
```

Comment:
I'm not sure why this is a todo, honestly.  Seems like selector and storage class annotation are mutually exclusive to me at this point.